### PR TITLE
Warn user if the docker config files doesn't contain expected information

### LINKED ISSUE #255: l0-setup: warn users when using dockercfg format
When l0-setup finds an empty `auth` section, throw an error.

### PR DESCRIPTION
**Pull request by sesh-kebab**
**What does this pull request do?**
Warns the user when running `l0-setup init` if the following is cases are met:

- if the auth property in the docker config file is empty
- if the docker config file is using a credential store

The warning message informs the user about the issue and suggests a potential solution.

**How should this be tested?**
Run l0-setup init with the docker config:

**Case 1: Missing Auth information**
```
{
	"auths": {}
}
```

**Case 2: Using a credential store**
```
{
	"auths": {
		"private-docker-repository.io": {}
	},
	"credsStore": "osxkeychain"
}
```

**Checklist**
- [ ] Unit tests
- [ ] Smoke tests (if applicable)
- [ ] System tests (if applicable)
- [ ] Documentation (if applicable)

closes #255

### CODE REVIEW

**Comment by sesh-kebab** [comment]
Please advise on styling of how to print really long, wrapping, sentences in go. I noticed other places had multiple calls to Printf for each line but seemed somewhat verbose to me.

Also, even though we are placing the burden of knowing the workings private docker repo authentication on our users, I feel like we can go further (either in our documentation or some other way). Needs more thought.

File: setup/docker/utils.go [hunk 1]
```diff
@@ -19,6 +19,30 @@ func LoadConfig(path string) (*Config, error) {
 
 	if config.Auths == nil {
 		config.Auths = map[string]Auth{}
+	} else if len(config.Auths) == 0 {
```
Comment (zpatrick) [inline setup/docker/utils.go]: This behavior should exist in the [createOrWriteDockerCFG](https://github.com/quintilesims/layer0/blob/develop/setup/instance/init.go#L100) function.

Comment (sesh-kebab) [inline setup/docker/utils.go]: Something that checks for an empty Auths object?

Comment (zpatrick) [inline setup/docker/utils.go]: yes - all of the added code basically

File: setup/instance/init.go [hunk 2]
```diff
@@ -123,5 +124,31 @@ func (l *LocalInstance) createOrWriteDockerCFG(dockerInputPath string) error {
...
+
+		fmt.Println("Press 'enter' to continue without private registry authentication: ")
+
+		var input string
+		fmt.Scanln(&input)
+	} else if config.CredsStore != "" {
+		fmt.Printf("[WARNING] Even though you have specified a path to a docker config file, "+
+			"the config file is using a credential store '%s' to cache the credentials. This "+
+			"means the credentials for private registry authentication aren't in the docker "+
+			"config file. You can either not use a credential store by removing the 'credsStore' "+
```
Comment (zpatrick) [inline setup/instance/init.go]: This wording is a little confusing. Maybe just simplify it with "Layer0 does not support credential store authentication."

Comment (sesh-kebab) [inline setup/instance/init.go]: How about:
> You are using a credential store '%s'. Layer0 does not support credential store authentication.

**Comment by zpatrick** [comment]
:+1:

**Pull request merged**